libandroid-navigation/src/main/java/org/maplibre/navigation/android/navigation/v5/location/engine/GoogleLocationEngineImpl.kt

```diff
@@ -68,10 +68,7 @@ open class GoogleLocationEngineImpl(
         )
     }
 
-    override fun removeLocationUpdates(listener: LocationCallback) {
-        // The listener is annotated with @NonNull, but there seems to be cases where a null
-        // listener is somehow passed into this method.
```

Comment:
So this is an issue of maplibre-native then? Should we create an issue there?

Reply:
I'm not sure it's a bug on the native side. As far as I can tell, the interface `org.maplibre.android.location.engine.LocationEngineImpl<T>::removeLocationUpdates(T listener)` doesn't make any promises regarding nullability of the `listener` parameter.
https://github.com/maplibre/maplibre-native/blob/1b87543af3f766a09115362372b9104f763ae6e5/platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/location/engine/LocationEngineImpl.java#L26

On the other hand, both this `GoogleLocationEngineImpl` and `AndroidLocationEngineImpl` in the native lib originally (ie. before my change) assumed that the `listener` is not null, so maybe there's something there 🤔 

For the sake of completness, here's the code in question: https://github.com/maplibre/maplibre-native/blob/1b87543af3f766a09115362372b9104f763ae6e5/platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/location/engine/AndroidLocationEngineImpl.java#L91

Reply:
Ok, thanks for clarifying. So we probably need to remove the non null annotation in Maplibre-Native as well. But then it's not really a bug there either, as Java can handle this stuff.

Reply:
damn, there was a comment that I ignored 👀 

Thanks for finding out!